Feature/vt bp review

cwickert opened the pull request:
I finished the review of the Virtualization Best practices. I covered a lot, but still don't think it's quite finished.

Open issues:
* inconsistent use of emphasis vs. literal
* inconsistent use of prompts and screens

cwickert commented:
![](https://github.trello.services/images/mini-trello-icon.png) [[DOCU] Virtualization Best Practices Guide](https://trello.com/c/bcm9xL2X/195-docu-virtualization-best-practices-guide)

fsundermeyer commented on xml/vt_best_practices.xml:
```diff
@@ -2805,7 +2800,8 @@ virtio-net-pci.any_layout=on/off
    </para>
    <para>
-    To use the <filename>hwrng</filename>, add the device in the XML
+    To use the <filename>hwrng</filename>, add an <literal>RNG</literal> device
```
Please explain RNG at the beginning of this section:

Virtio RNG (random number generator) is a paravirtualized device ...

and then replace

<filename>hwrng</filename> with "Virtio RNG"

cwickert replied:
Done

fsundermeyer commented on xml/vt_best_practices.xml:
```diff
@@ -2950,6 +2946,8 @@ virtio_rng.0</screen>
   <sect2 xml:id="vt.best.perf.mtype">
    <title>Updating the Guest Machine Type</title>
+   <!-- cwickert 2017-01-26: The sentences in the next two paragaphs are quite
+    long and hard to read/understand. Try to make them shorter? -->
```
&qemu; machine types define details of the architecture that are particularly relevant during migration and session management. The list of machine types is constantly maintained&mdash;the addition of new features and fixes results in new machine types.
Old machine types remain in the list for compatibility reasons when new versions are released. However, to take advantage of improvements, it is recommended to always migrate to the latest machine type when upgrading.

cwickert replied:
I've use a yer shorter combination of the original and your suggestion. Thanks!

fsundermeyer commented on xml/vt_best_practices.xml:
```diff
@@ -2987,7 +2985,7 @@ virtio_rng.0</screen>
     The ability to change a &vmguest;'s state heavily depends on the
     operating system. It is very important to test this feature before any
-    use of your &vmguest; in production. For example most Linux operating
+    use of your &vmguest; in production. For example, most Linux operating
```
...befor using your &vmguest;s...

cwickert replied:
Fixed

fsundermeyer commented on xml/vt_best_practices.xml:
```diff
@@ -3052,23 +3049,22 @@ error: Failed to start domain sles12
-    changed on a per &vmguest; basis, use <command>virsh edit</command> to
-    add the listen XML attribute to the <emphasis>graphics
+    To fix this, you can change the default <emphasis>spice_listen</emphasis>
+    value in <filename>/etc/libvirt/qemu.conf</filename> using the local ipv6
```
&lt;literal>spice_listen&lt;/literal>

cwickert replied:
Fixed

fsundermeyer commented on xml/vt_best_practices.xml:
```diff
@@ -3052,23 +3049,22 @@ error: Failed to start domain sles12
-    add the listen XML attribute to the <emphasis>graphics
+    To fix this, you can change the default <emphasis>spice_listen</emphasis>
+    value in <filename>/etc/libvirt/qemu.conf</filename> using the local ipv6
+    address <emphasis>::1</emphasis>. The spice server listening address can
```
&lt;literal>1&lt;/literal>

cwickert replied:
<systemitem class="ipaddress">::1</systemitem>

cwickert replied:
&lt;systemitem class="ipaddress"&gt;::1&lt;/systemitem&gt;

fsundermeyer commented on xml/vt_best_practices.xml:
```diff
@@ -3052,23 +3049,22 @@ error: Failed to start domain sles12
-    &vmguest; from the XML file. Use <command>virsh</command> with
-    <option>domxml</option>.
+    &vmguest; from the XML file. Use <command>virsh</command> with the
+    <option>domxml</option> option.
```
I'd rather delete this sentence

cwickert replied:
Done.

fsundermeyer commented on xml/vt_best_practices.xml:
```diff
@@ -3828,7 +3824,7 @@ VGA trouble debug
       Run the sandbox, using <command>virt-sandbox</command>. This command
       has many interesting options, read its man page to discover them all.
-      It can either be running as root or as an unprivileged user.
+      The command can be run as root or as an unprivileged user.
```
...&rootuser;...

cwickert replied:
Fixed.

fsundermeyer commented on xml/vt_best_practices.xml:
```diff
@@ -3879,14 +3876,14 @@ VGA trouble debug
   </para>
   <para>
-   First you need to change to the <emphasis>-default</emphasis> kernel, if not
+   First you need to change to the <emphasis>-default</emphasis> kernel; if not
```
First you need to change to the default Kernel (package <package>kernel-default</papckage>); if not
￼already installed, you need to install it while in PV mode.

_Note: No "must", it should rather be "need to" (according to the Styleguide)._

cwickert replied:
Fixed.

cwickert commented:
Thanks for the review! IFAICS I addressed all of your points (and many more). I've also spell-, link, and -stylechecked everything.

The only remaining warnings/erros from stylechecker are

* some long sentences
* missing introductory paragraphs or abstracts on some sections. IHMO this is ok for a best practice.
* > Deliverable article.vt.best.practices does not contain a date.
  >       
  >      Add a date element, use <?dbtimestamp?> inside.
  Do we add dates to Best Practices?

fsundermeyer commented:
Yes, I think we should add a date - it will tell people when the guide has been updated.

fsundermeyer commented:
Everything is fine now, thanks a lot!
